docs/package.json

```diff
@@ -83,7 +83,7 @@
     "react-markdown": "^8.0.3",
     "sitemap-urls": "^3.0.0",
     "ts-jest": "^27",
-    "ts-morph": "^15.1.0",
+    "ts-morph": "^17.0.0",
```

Comment:
We need to use ts-morph 17 because this PR bumps monorepo `typescript` to `4.9`, and only `ts-morph` 17 is compatible with `4.9`.

Reading through their [changelog](https://github.com/dsherret/ts-morph/blob/latest/packages/ts-morph/CHANGELOG.md), their only breaking change was bumping `typescript` on their end.

Reply:
Where do we bump the monorepo `typescript`? Is it only in the yarn.lock?

Reply:
It's from our monorepo package.json: 

https://github.com/aws-amplify/amplify-ui/blob/0da924cd5ee2884b89b0e184ab38bbd9fb57327a/package.json#L95

Since we use caret there, this gets bumped as we refresh the lockfile. This will be visible in `yarn.lock`.